Description: Add autotools based build rules to get a build system
 that link with the libraries included in Debian instead of trying to
 wget and run source tarballs from around the Internet.
Forwarded: forwarded via email to upstream
Author: Petter Reinholdtsen <pere@hungry.com>
Last-Update: 2015-06-01

Index: bs1770gain-git/Makefile.am
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ bs1770gain-git/Makefile.am	2015-06-01 21:25:25.000000000 +0000
@@ -0,0 +1,3 @@
+ACLOCAL_AMFLAGS = -I m4
+
+SUBDIRS = libpbutil lib1770-2 libffsox-2  bs1770gain examples
Index: bs1770gain-git/bs1770gain/Makefile.am
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ bs1770gain-git/bs1770gain/Makefile.am	2015-06-01 21:33:52.314296949 +0000
@@ -0,0 +1,30 @@
+bin_PROGRAMS = \
+	bs1770gain
+
+bs1770gain_LDADD = \
+	-L../lib1770-2 \
+	-L../libffsox-2 \
+	-L../libpbutil \
+	-l1770-2 \
+	-lffsox-2 \
+	-lpbutil \
+	-lavformat-ffmpeg \
+	-lavutil-ffmpeg \
+	-lsox \
+	-lm
+
+bs1770gain_SOURCES = \
+	bs1770gain_aggregate.c \
+	bs1770gain_album.c \
+	bs1770gain.c \
+	bs1770gain.h \
+	bs1770gain_opath.c \
+	bs1770gain_parse_time.c \
+	bs1770gain_priv.h \
+	bs1770gain_transcode.c \
+	bs1770gain_tree.c
+
+AM_CFLAGS= \
+        -I$(top_srcdir)/libpbutil \
+        -I$(top_srcdir)/libffsox-2 \
+        -I$(top_srcdir)/lib1770-2
Index: bs1770gain-git/configure.ac
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ bs1770gain-git/configure.ac	2015-06-01 21:25:25.000000000 +0000
@@ -0,0 +1,41 @@
+# -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ(2.57)
+AC_INIT([bs1770gain],[0.4.1],[pbelkner (at) snafu.de])
+AM_SILENT_RULES([yes])
+AM_INIT_AUTOMAKE([foreign 1.9 -Wall])
+AC_CONFIG_MACRO_DIR([m4])
+
+AC_PROG_CC
+AM_PROG_AR
+
+LT_INIT
+
+if test "$GCC" = "yes"; then
+  for flag in \
+      -Wall \
+      -Wformat \
+      -Wpointer-arith \
+      -Wcast-align \
+      -Wcast-qual \
+      -Wreturn-type \
+      -Wextra \
+      -Wstrict-prototypes \
+      -Wmissing-declarations \
+      -Wmissing-prototypes
+#      -Werror
+  do
+    JAPHAR_GREP_CFLAGS($flag, [ CFLAGS="$CFLAGS $flag" ])
+  done
+fi
+
+AC_CONFIG_FILES([
+	Makefile
+	libpbutil/Makefile
+	lib1770-2/Makefile
+	libffsox-2/Makefile
+	bs1770gain/Makefile
+	examples/Makefile
+])
+AC_OUTPUT
Index: bs1770gain-git/examples/Makefile.am
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ bs1770gain-git/examples/Makefile.am	2015-06-01 21:25:25.000000000 +0000
@@ -0,0 +1 @@
+EXTRA_DISTS = COPYING ffmpeg1770.c  sndfile1770.c
Index: bs1770gain-git/lib1770-2/Makefile.am
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ bs1770gain-git/lib1770-2/Makefile.am	2015-06-01 21:25:25.000000000 +0000
@@ -0,0 +1,13 @@
+lib_LTLIBRARIES = lib1770-2.la
+lib1770_2_la_LIBADD = \
+	-lm
+lib1770_2_la_SOURCES = \
+	lib1770_biquad.c \
+	lib1770_block.c \
+	lib1770.h \
+	lib1770_pre.c \
+	lib1770_stats.c
+
+AM_CFLAGS= \
+        -I$(top_srcdir)/libpbutil \
+        -I$(top_srcdir)/libffsox-2
Index: bs1770gain-git/libffsox-2/Makefile.am
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ bs1770gain-git/libffsox-2/Makefile.am	2015-06-01 21:25:25.000000000 +0000
@@ -0,0 +1,58 @@
+lib_LTLIBRARIES = libffsox-2.la
+
+libffsox_2_la_LIBADD = \
+	-L../lib1770-2 \
+	-L../libpbutil \
+	-lavformat-ffmpeg \
+	-lavcodec-ffmpeg \
+	-lavutil-ffmpeg \
+	-lsox \
+	-l1770-2 \
+	-lpbutil \
+	-lm
+
+libffsox_2_la_SOURCES = \
+	ffsox_aggregate.c \
+	ffsox_analyze.c \
+	ffsox_audio_player.c \
+	ffsox_audiostream.c \
+	ffsox_collect.c \
+	ffsox_convert.c \
+	ffsox_csv2avdict.c \
+	ffsox_dynload.c \
+	ffsox_dynload.h \
+	ffsox_find_decoder.c \
+	ffsox_frame.c \
+	ffsox_frame_consumer.c \
+	ffsox_frame_convert.c \
+	ffsox_frame_convert_sox.c \
+	ffsox_frame_reader.c \
+	ffsox_frame_writer.c \
+	ffsox.h \
+	ffsox_machine.c \
+	ffsox_msvcrt.c \
+	ffsox_node.c \
+	ffsox_packet_consumer.c \
+	ffsox_packet_consumer_list.c \
+	ffsox_packet_writer.c \
+	ffsox_path3.c \
+	ffsox_priv.h \
+	ffsox_sink.c \
+	ffsox_source.c \
+	ffsox_source_link.c \
+	ffsox_source_progress.c \
+	ffsox_sox_add_effect.c \
+	ffsox_sox_pull_handler.c \
+	ffsox_sox_reader.c \
+	ffsox_sox_read_handler.c \
+	ffsox_stream.c
+#	mux1.c \
+#	mux2.c \
+#	mux3.c
+
+#	wasapi.c
+#	player.c
+
+AM_CFLAGS= \
+        -I$(top_srcdir)/libpbutil \
+        -I$(top_srcdir)/lib1770-2
Index: bs1770gain-git/libpbutil/Makefile.am
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ bs1770gain-git/libpbutil/Makefile.am	2015-06-01 21:25:25.000000000 +0000
@@ -0,0 +1,20 @@
+lib_LTLIBRARIES = libpbutil.la
+libpbutil_la_SOURCES = \
+	pbu_basename.c \
+	pbu_copy_file.c \
+	pbu_ext.c \
+	pbu_extend_path.c \
+	pbu_list.c \
+	pbu_mkdir.c \
+	pbu_msvcrt.c \
+	pbu_s2w.c \
+	pbu_same_file.c \
+	pbu_strtok.c \
+	pbutil.h \
+	pbutil_priv.h \
+	pbu_wcstok.c
+
+AM_CFLAGS= \
+        -I$(top_srcdir)/libpbutil \
+        -I$(top_srcdir)/libffsox-2 \
+        -I$(top_srcdir)/lib1770-2
Index: bs1770gain-git/m4/japhar.m4
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ bs1770gain-git/m4/japhar.m4	2015-06-01 21:25:25.000000000 +0000
@@ -0,0 +1,20 @@
+dnl
+dnl Add macros
+dnl     JAPHAR_GREP_CFLAGS
+dnl
+
+dnl
+dnl JAPHAR_GREP_CFLAGS(flag, cmd_if_missing, cmd_if_present)
+dnl
+dnl From Japhar.
+dnl
+AC_DEFUN([JAPHAR_GREP_CFLAGS],
+[case "$CFLAGS" in
+"$1" | "$1 "* | *" $1" | *" $1 "* )
+  ifelse($#, 3, [$3], [:])
+  ;;
+*)
+  $2
+  ;;
+esac
+])
Index: bs1770gain-git/README
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ bs1770gain-git/README	2015-06-01 21:25:25.000000000 +0000
@@ -0,0 +1,14 @@
+BS1770GAIN
+==========
+
+BS1770GAIN is a loudness scanner implementing ITU-R BS.1770.
+
+The latest version is available from http://bs1770gain.sourceforge.net/ .
+
+Report bugs and improvements to the sourceforge project.
+
+To build it from the git repo, run
+
+  autoreconf -i
+  ./configure
+  make
